packages/web/modals/review-order.tsx

```diff
@@ -0,0 +1,686 @@
+import { isValidNumericalRawInput } from "~/hooks/input/use-amount-input";
+import { useSwap } from "~/hooks/use-swap";
+import { ModalBase } from "~/modals";
+import { formatPretty, getPriceExtendedFormatOptions } from "~/utils/formatter";
+
+interface ReviewOrderProps {
+  isOpen: boolean;
+  onClose: () => void;
+  confirmAction: () => void;
+  isConfirmationDisabled: boolean;
+  slippageConfig?: ObservableSlippageConfig;
+  outAmountLessSlippage?: IntPretty;
+  outFiatAmountLessSlippage?: PricePretty;
+  outputDifference?: RatePretty;
+  showOutputDifferenceWarning?: boolean;
+  percentAdjusted?: Dec;
+  limitPriceFiat?: PricePretty;
+  limitSetPriceLock?: (lock: boolean) => void;
+  baseDenom?: string;
+  title: string;
+  gasAmount?: PricePretty;
+  isGasLoading?: boolean;
+  gasError?: Error | null;
+  expectedOutput?: CoinPretty;
+  expectedOutputFiat?: PricePretty;
+  inAmountToken?: CoinPretty;
+  inAmountFiat?: PricePretty;
+  fromAsset?: ReturnType<typeof useSwap>["fromAsset"];
+  toAsset?: ReturnType<typeof useSwap>["toAsset"];
+  page?: EventPage;
+  isBeyondOppositePrice?: boolean;
+}
+
+export function ReviewOrder({
+  isOpen,
+  onClose,
+  confirmAction,
+  isConfirmationDisabled,
+  slippageConfig,
+  outAmountLessSlippage,
+  outFiatAmountLessSlippage,
+  outputDifference,
+  showOutputDifferenceWarning,
+  percentAdjusted,
+  limitPriceFiat,
+  baseDenom,
+  title,
+  gasAmount,
+  isGasLoading,
+  gasError,
+  limitSetPriceLock,
+  expectedOutput,
+  expectedOutputFiat,
+  inAmountToken,
+  inAmountFiat,
+  toAsset,
+  fromAsset,
+  page,
+  isBeyondOppositePrice = false,
+}: ReviewOrderProps) {
+  const { t } = useTranslation();
+  // const { isMobile } = useWindowSize();
+  const { logEvent } = useAmplitudeAnalytics();
+  const [manualSlippage, setManualSlippage] = useState("");
+  const [isEditingSlippage, setIsEditingSlippage] = useState(false);
+  const isManualSlippageTooHigh = useMemo(
+    () => +manualSlippage > 1,
+    [manualSlippage]
+  );
+  const [tab] = useQueryState("tab", parseAsString.withDefault("swap"));
+
+  const { isOneClickTradingEnabled } = useOneClickTradingSession();
+
+  const isManualSlippageTooLow = useMemo(
+    () => manualSlippage !== "" && +manualSlippage < 0.1,
+    [manualSlippage]
+  );
+
+  const [orderType] = useQueryState(
+    "type",
+    parseAsString.withDefault("market")
+  );
+
+  const initialOutput = useMemo(
+    () => outAmountLessSlippage ?? new IntPretty(0),
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+    []
+  );
+
+  const { diffGteSlippage, restart } = useMemo(
+    () => {
+      let originalValue = initialOutput;
+      return {
+        diffGteSlippage: slippageConfig
+          ? originalValue
+              .sub(outAmountLessSlippage ?? new IntPretty(0))
+              .toDec()
+              .gte(slippageConfig?.slippage.toDec())
+          : false,
+        restart: () => {
+          originalValue = outAmountLessSlippage ?? new IntPretty(0);
```

Comment:
This seems like a state change? Perhaps we should use a `useState` hook or something to ensure re-renders when state changes?

Reply:
Fabrizio wanted to use `useMemo` to ensure that it does not rerender unintentionally. This value is supposed to be static once the component has rendered and is used for comparison.

Reply:
OK if it's just for comparison, that's fine